sdk/resources/mgmt/src/main/java/com/azure/management/resources/fluentcore/utils/PagedList.java

```diff
@@ -0,0 +1,367 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.management.resources.fluentcore.utils;
+
+import com.azure.core.http.rest.PagedIterable;
+import com.azure.core.http.rest.PagedResponse;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.ConcurrentModificationException;
+import java.util.Iterator;
+import java.util.List;
+import java.util.ListIterator;
+import java.util.NoSuchElementException;
+import java.util.Objects;
+
+/**
+ * Defines a list response from a paging operation. The pages are
+ * lazy initialized when an instance of this class is iterated.
+ *
+ * @deprecated Use {@link PagedIterable} instead.
+ *
+ * @param <E> the element type.
+ */
+@Deprecated
+public final class PagedList<E> implements List<E> {
+
+    // The invariant is that before / after read-only method, items + pagedResponseIterator is the complete pagedIterable.
+
+    /** The items retrieved. */
+    private final List<E> items;
+
+    /** The paged response iterator for not retrieved items. */
+    private Iterator<PagedResponse<E>> pagedResponseIterator;
+
+    /**
+     * Creates an instance of PagedList.
+     */
+    public PagedList() {
+        items = new ArrayList<>();
+        pagedResponseIterator = Collections.emptyIterator();
+    }
+
+    /**
+     * Creates an instance of PagedList from a {@link PagedIterable}.
+     *
+     * @param pagedIterable the {@link PagedIterable} object.
+     */
+    public PagedList(PagedIterable<E> pagedIterable) {
+        items = new ArrayList<>();
+        Objects.requireNonNull(pagedIterable, "'pagedIterable' cannot be null.");
+        this.pagedResponseIterator = pagedIterable.iterableByPage().iterator();
+    }
+
+    /**
+     * If there are more pages available.
+     *
+     * @return true if there are more pages to load. False otherwise.
+     */
+    protected boolean hasNextPage() {
+        return pagedResponseIterator.hasNext();
+    }
+
+    /**
+     * Loads a page from next page link.
+     * The exceptions are wrapped into Java Runtime exceptions.
+     */
+    protected void loadNextPage() {
+        this.items.addAll(pagedResponseIterator.next().getValue());
```

Comment:
seems cachedpage removed, any concern on performance etc when original cachedpage adding? Or we may revisit it later when there's complain

Reply:
v2 should not use PagedList. This class is provided since return type in v2 is PagedIterable, which is not compatibly with v1 PagedList.

Reply:
PagedIterable/PagedFlux is already doing what need to be done on paged response. This class is only a wrapper to provide interface compatible with v1.

Reply:
With this, user now have 2 choice when migrate.
1. Use PagedIterable, and refactor any code that handling it.
2. Use PagedList, only change the return type. Find time later to do all refactor needed in 1.

Reply:
thanks for clarification. Then in our sample or migration guide, pls use PagedIterable as right and long term way for paging operation.